### PR TITLE
Notes integration

### PR DESCRIPTION
**Pull request by lastminutediorama**

### CODE REVIEW

**Review by pflopez — COMMENTED** [review]
I think this is great! Having an abstract note service will make our life easier for sure.
We probably want to simplify the concrete implementations so they don't need to redefine the add/list/remove methods though. Kind of defeats the purpose.

This may also be simplified:

```
protected multipleUrl = (objectId: string) =>
    `/planning/planning_area/${objectId}/note`;
  protected singleUrl = (objectId: string, noteId: string) =>
    `/planning/planning_area/${objectId}/note/${noteId}`;
```

Maybe we could have something like 
```
baseUrl = "/planning/planning_area/"
multipleUrl = this.baseUrl || options. // optional, should actually be the same
parentResourceId = this.planningAreaId
resource = '/note'
```

So when the abstract service constructs the urls can do 
```
  getNotes(objectId: string | number) {
    objectId = objectId.toString();
    return this.http.get<Note[]>(
      environment.backend_endpoint.concat(this.baseUrl + parentResourceId + resource),
      {
        withCredentials: true,
      }
    );
  }
  ```

I see that the project area node endpoint requires this:
```
params: { project_area_pk: objectId.toString() },
```

I'll say that's wrong on the API side. We are already proving the ID on the url, so its redundant.

File: src/interface/src/app/services/notes.service.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+}
+@Injectable()
+export class ProjectAreaNotesService extends BaseNotesService {
+  protected modelName: NotesModelName = 'planning_area';
+  protected multipleUrl = () => `/v2/projectarea-notes/`;
+
+  protected singleUrl = (objectId: string, noteId: string) =>
+    `/project_area_note/${objectId}/note/${noteId}`;
+
+  override getNotes(objectId: number | string): Observable<Note[]> {
```
[the remark this answers is not in the document]
Comment (lastminutediorama) [reply src/interface/src/app/services/notes.service.ts:76]: I meant to follow up on this. A major underlying issue here is that we have a older, nested route for planning area notes, as well as an unnested route for project areas, so unfortunately - despite my desire to just use one abstract service here, it won't work for both situations.

I think ideally, we would change the planning area endpoints to be unnested, but we also have a feature flagged version of notes, so I'm not sure if it makes sense to do that now, or wait until a future version to change the backend. I'm open to whatever makes sense.

File: src/planscape/planning/urls.py [hunk 2]
```diff
@@ -109,7 +109,7 @@
     ),
     # Planning Area Notes C/R/D
     path(
-        "planning_area/<int:planningarea_pk>/note",
```
Comment (lastminutediorama) [inline src/planscape/planning/urls.py:112]: :disappointed:

**Review by pflopez — APPROVED** [review]
Nice! LGTM! 🙌

File: src/interface/src/app/treatments/treatment-project-area/treatment-project-area.component.ts [hunk 3]
```diff
@@ -51,4 +67,86 @@ export class TreatmentProjectAreaComponent implements OnDestroy {
     this.selectedStandsState.clearStands();
     this.treatmentsState.setShowApplyTreatmentsDialog(false);
   }
+
+  treatmentPlanId: number = this.route.snapshot.data['treatmentId'];
```
Comment (pflopez) [inline src/interface/src/app/treatments/treatment-project-area/treatment-project-area.component.ts:71]: ah- forgot to mention this but you should use the treatmentState instead for this? like we are already handling this stuff there. Not a blocker but that way we dont couple this component with route stuff.

File: src/interface/src/app/treatments/treatment-project-area/treatment-project-area.component.ts [hunk 4]
```diff
@@ -51,4 +67,86 @@ export class TreatmentProjectAreaComponent implements OnDestroy {
...
+      });
+  }
+
+  loadNotes() {
+    if (this.projectAreaId) {
+      this.notesService
+        .getNotes(this.projectAreaId)
+        .subscribe((notes: Note[]) => {
+          this.notes = notes;
+          this.cdr.detectChanges();
```
Comment (pflopez) [inline src/interface/src/app/treatments/treatment-project-area/treatment-project-area.component.ts:148]: also forgot to mention, but why do you need this.cdr.detectChanges();? this is usually a tell-tale of something else being wrong, unless you are using dependencies that run outside angular cycle, which this one doesnt seem to be the case?

Comment (lastminutediorama) [reply src/interface/src/app/treatments/treatment-project-area/treatment-project-area.component.ts]: This was intermittently slow. I don't think changes to `this.notes` was forcing a redraw, but using spread seemed to resolve it.

**Pull request merged**